frost-core/src/frost/keys.rs

```diff
@@ -156,11 +178,58 @@ pub(super) struct CoefficientCommitment<C: Ciphersuite>(pub(super) Element<C>);
 /// [`VerifiableSecretSharingCommitment`], either by performing pairwise comparison, or by using
 /// some agreed-upon public location for publication, where each participant can
 /// ensure that they received the correct (and same) value.
-#[derive(Clone)]
+#[derive(Clone, PartialEq)]
 pub struct VerifiableSecretSharingCommitment<C: Ciphersuite>(
-    pub(super) Vec<CoefficientCommitment<C>>,
+    pub(crate) Vec<CoefficientCommitment<C>>,
 );
 
+impl<C> VerifiableSecretSharingCommitment<C>
+where
+    C: Ciphersuite,
+{
+    /// Returns concatenated coefficent commitments from self.0 specifying the number of commitments
+    pub fn serialize(&self) -> String {
+        let values = self
+            .0
+            .iter()
+            .map(|cc| hex::encode(<<C as Ciphersuite>::Group as Group>::serialize(&cc.0)))
+            .join("");
+
+        hex::encode(self.0.len().to_string()) + &values
+    }
```

Comment:
We shouldn't force a particular serialization format for lists and we need to allow the caller to use their own format.

To do that we can change this to return just a vector of serialized elements, and the caller can then serialize the vector as they wish.

Sorry I didn't make this clearer! But we can still use the vector-serialization strategy you used here in the demo (where we can encode things any way we want)

```suggestion
    pub fn serialize(&self) -> Vec<<C::Group as Group>::Serialization> {
        self.0
            .iter()
            .map(|cc| <<C as Ciphersuite>::Group as Group>::serialize(&cc.0))
            .collect()
    }
```